language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/QNameRefCompletionUtil.java

```diff
@@ -70,7 +69,7 @@ public static List<Symbol> getExpressionContextEntries(BallerinaCompletionContex
                         || symbol.kind() == SymbolKind.TYPE_DEFINITION
                         || symbol.kind() == SymbolKind.CLASS
                         || symbol instanceof VariableSymbol)
-                .collect(Collectors.toList())).orElseGet(ArrayList::new);
+                .toList()).orElseGet(ArrayList::new);
```

Comment:
Let's make both branches return immutable lists to avoid any suprises
```suggestion
                .collect(Collectors.toList())).orElseGet(Collections::emptyList);
```

Reply:
Your suggestion would create a mutable list if it is non-empty. 
I changed it to `.toList()).orElseGet(Collections::emptyList)`.
Fixed in [2778660](https://github.com/ballerina-platform/ballerina-lang/pull/42904/commits/277866060d74b08115089625bb7d582c095697a1)